OneHot.lua

```diff
@@ -9,12 +9,22 @@ function OneHot:__init(outputSize)
 end
 
 function OneHot:updateOutput(input)
-   local size = input:size():totable()
+   local size
+   if type(input) == 'number' then
+      if self:type() == 'torch.CudaTensor' then
+			input = torch.CudaTensor({input})
```

Comment:
Could you use a lazy-initialized buffer here instead of allocating memory every call to forward?
Example:
```lua
self._input = self._input or torch.CudaTensor()
self._input:resize(1)[1] = input
input = self._input
```
You should also make sure it is set to nil in` OneHot:type()` to prevent weird casting bugs.